### PR TITLE
How-To: Configure Monitoring Alerts using Alertmanager

### PR DESCRIPTION
**Pull request by bastjan**
This PR adds samples for Slack, Telegram, and Email. It also tries to explain routing and adds a test and debug section.

I'll expand the debug section with a webhook tester in an additional PR.

## Checklist

- [x] Try to isolate changes into separate PRs (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `how-to`, `tutorial`, `reference`, `explanation`, `cicd`, `dependency`
      as they show up in the changelog

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+This makes it difficult to debug issues with Alertmanager.
+The following section describes some common issues and how to debug them.
+
+. Check all secret references.
++
+One missing secret reference blocks the creation of all the receivers in the `AlertmanagerConfig`.
+
+. Check the secret values.
++
+Wrong login credentials or missing certificates can cause the Alertmanager to fail to send alerts.
```
Comment (bastjan) [inline docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc:445]: Will add a debug your webhook section in a separate PR.

File: docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc [hunk 2]
```diff
@@ -0,0 +1,448 @@
+= Configure Alerts using Alertmanager
+
+{product} allows you to configure alerts receivers to notify you when an alert is triggered.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc:3]:
```suggestion
{product} allows you to configure custom alert receivers to notify you when an alert is triggered.
```

File: docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+Might be omitted for LOGIN protocol.
+<6> The secret created earlier.
+
+. Test the email receiver. See <<test-alerting>>.
+
+== Routing Alerts [[routing]]
+
+Alerts can be routed to different receivers based on the labels attached to the alerts.
+
+A `AlertmanagerConfig` resource defines a routing tree with a root route and child routes.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc:291]:
```suggestion
An `AlertmanagerConfig` resource defines a routing tree with a root route and child routes.
```

File: docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+This makes it difficult to debug issues with Alertmanager.
+The following section describes some common issues and how to debug them.
+
+. Check all secret references.
++
+One missing secret reference blocks the creation of all the receivers in the `AlertmanagerConfig`.
+
+. Check the secret values.
++
+Wrong login credentials or missing certificates can cause the Alertmanager to fail to send alerts.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc:448]:
```suggestion
Alertmanager may fail to send alerts when the routing configuration has wrong login credentials or missing certificates.
```

File: docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+[source,bash]
+----
+kubectl -n "${APP_NAMESPACE}" delete prometheusrule test-alert
+----
+
+. Repeat the test for all receivers by adding the appropriate labels to the alert.
+
+== Troubleshooting [[troubleshooting]]
+
+OpenShift currently doesn't expose any logs or status information for the Alertmanager.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc:438]:
```suggestion
OpenShift currently doesn't expose any logs or status information for Alertmanager.
```

**Review by simu — APPROVED** [review]
LGTM overall, some wording suggestions inline

**Review by corvus-ch — CHANGES_REQUESTED** [review]
This feels like too many things cramped into one thing. This probably will confuse readers. I suggest splitting this up into different documents: one for each receiver.

While it is nice to talk about the routing capabilities, I would reduce this. Maybe go as far as to reduce this to a tip with a link to upstream documentation.

File: docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+  - name: out-of-office-hours <3>
+    emailConfigs:
+    - to: alerts@example.com
+      from: my-app@example.com
+      smarthost: smtp.example.com:587
+      authUsername: my-app@example.com
+      authIdentity: my-app@example.com
+      authPassword:
+        key: password
+        name: email
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc:50]:
```suggestion
        name: email-alert-auth
```

**Review by corvus-ch — CHANGES_REQUESTED** [review]

**Comment by bastjan** [comment]
@corvus-ch I shrunk down the initial sample and re-checked and improved manifest naming.

The initial sample is now more a ToC and helps guide the user to the correct section.

File: docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc (hunk 1 above)
Comment (bastjan) [reply docs/modules/ROOT/pages/how-to/configure-alertmanager.adoc]: Added in https://github.com/appuio/appuio-cloud-docs/pull/141/commits/7618e3135fe88b508320d44955d82f88daf5f19b

**Review by corvus-ch — APPROVED** [review]

**Pull request merged**